pkg/controller/dnszone/azureactuator.go

```diff
@@ -85,7 +89,8 @@ func (a *AzureActuator) Delete() error {
 	resourceGroupName := a.dnsZone.Spec.Azure.ResourceGroupName
 	logger := a.logger.WithField("zone", a.dnsZone.Spec.Zone)
 
-	if err := a.deleteRecordsets(logger); err != nil {
+	logger.Info("Deleting recordsets in managedzone")
+	if err := DeleteAzureRecordSets(a.azureClient, a.dnsZone, logger); err != nil {
```

Comment:
Be consistent between using `a.dnsZone.Status.Azure` and `a.managedZone`.

Reply:
For AWS and GCP we actually store some information about the DNS Zone (the ZoneID and ZoneName respectively). For Azure, the dnszone.Spec.Name **is** the name of the object up in Azure. I originally wrote up some code to add a status field for Azure DNSZones, but it wasn't storing anything useful. WDYT?

Reply:
No, don't put anything in the status. My original comment should have said "be consistent between using `a.dnsZone.Spec.Zone` and `a.managedZone`". Like with the GCP PR, I want you to use `a.dnsZone.Spec.Zone` everywhere in the `Delete` function rather than using `*a.managedZone.Name` in some places.